app/scenes/channel/channel.js

```diff
@@ -93,35 +102,25 @@ export default class Channel extends React.Component {
                         onCloseStart={this.closeRightSidebar}
                         openDrawerOffset={0.2}
                     >
-                        <View style={{backgroundColor: theme.sidebarHeaderBg, flexDirection: 'row', justifyContent: 'flex-start', marginTop: 20}}>
-                            <View style={{flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
-                                <TouchableHighlight
-                                    onPress={this.props.actions.openChannelDrawer}
-                                    style={{height: 25, width: 25, marginLeft: 10, marginRight: 10}}
-                                >
-                                    <Icon
-                                        name='bars'
-                                        size={25}
-                                        color={theme.sidebarHeaderTextColor}
-                                    />
-                                </TouchableHighlight>
-                            </View>
-                            <View style={{flex: 1, flexDirection: 'row', justifyContent: 'flex-start', alignItems: 'center', backgroundColor: theme.sidebarHeaderBg}}>
-                                <Text style={{color: theme.sidebarHeaderTextColor, fontSize: 15, fontWeight: 'bold'}}>
-                                    {currentChannel.display_name}
-                                </Text>
-                            </View>
-                            <TouchableHighlight
-                                onPress={this.openRightSidebar}
-                                style={{height: 50, width: 50}}
-                            >
-                                <Text style={{color: theme.sidebarHeaderTextColor}}>{'>'}</Text>
-                            </TouchableHighlight>
-                        </View>
+                        <ChannelHeader
+                            currentChannel={currentChannel}
+                            openLeftSidebar={this.props.actions.openChannelDrawer}
+                            openRightSidebar={this.openRightSidebar}
+                        />
                         <ChannelPostList channel={currentChannel}/>
+                        <PostTextbox
+                            value={this.state.valuetodo}
+                            teamId={currentChannel.team_id}
+                            channelId={currentChannel.id}
+                            onChangeText={this.onChangeTextTodo}
+                        />
                     </Drawer>
                 </ChannelDrawer>
-            </View>
+            </KeyboardAvoidingView>
         );
     }
+
+    onChangeTextTodo = (text) => {
```

Comment:
Oh right, this was supposed to go into the view state. I knew there was something else I meant to do before submitting this PR